calcom/cal.com
# feat: Overlay Calendar v2 and Troubleshooter v2

## What does this PR do?

- This PR currently only fetches event titles for Google Calendar.
- This PR allows users to use Overlay Calendar feature by signing in using their Google Account.
- It also pulls actual event title from the google calendar instead of just showing "Busy" for both overlay calendar and troubleshooter.
- This feature upgrade does not need any Google scope changes.

Fixes #12763  
Fixes #12752 

Overlay Calendar v2
https://www.loom.com/share/8e3889f4f7034596adb2bfdc640cba11?sid=7242ef08-ef5e-4b9a-a102-c19eded1d882

Troubleshooter v2
https://www.loom.com/share/8024891797574a748b99c3c40979398c?sid=248174de-8abe-4fd3-a42e-c9146b86ddf1

## Type of change



- New feature (non-breaking change which adds functionality)
- This change requires a documentation update:
  - Under Authorized redirect URI's in google cloud add the URI: `<Cal.com URL>/api/auth/callback/google-for-overlay`
replacing Cal.com URL with the URI at which your application runs.


## How should this be tested?

- Please make sure to obtain google cloud API Credentials as mentioned [here](https://github.com/calcom/cal.com?tab=readme-ov-file#obtaining-the-google-api-credentials).
- No new scopes required.
- Edit the credential in the google cloud.
- Under Authorized redirect URI's in google cloud add the URI: `<Cal.com URL>/api/auth/callback/google-for-overlay`
replacing Cal.com URL with the URI at which your application runs.

### Testing Overlay Calendar v2
- Just go to a booking page and toggle the Overlay Calendar switch.
- Choose Continue with Google and sign in with a google id.
- You will see that now event titles are being displayed instead of just "Busy" for the calendar busy events.

### Testing Troubleshooter v2
- Just go to the Troubleshooter and make sure that google calendar is connected and the app is checking google calendar for conflicts. 
- You will see that now event titles are being displayed instead of just "Busy" for the calendar busy events.

## Mandatory Tasks
